src/Controller/Admin/UserController.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller\Admin;
+
+use App\Entity\User;
+use App\Form\UserUpdateFormType;
+use App\Repository\UserRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class UserController extends AbstractController
+{
+    private UserRepository $userRepository;
+
+    private EntityManagerInterface $entityManager;
+
+    public function __construct(UserRepository $userRepository, EntityManagerInterface $entityManager)
+    {
+        $this->userRepository = $userRepository;
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * @Route(path="/users", methods={"GET"}, name="list_users")
+     */
+    public function listAllUsers(): Response
+    {
+        $users = $this->userRepository->findAll();
+
+        return $this->render('admin/listUsers.html.twig', [
+            'users' => $users,
+        ]);
+    }
+
+    /**
+     * @Route(path="/users/{id}", methods={"GET", "POST"}, name="update_user")
+     */
+    public function updateUserAction(int $id, Request $request): Response
+    {
+        $user = new User();
+
+        $form = $this->createForm(UserUpdateFormType::class, $user);
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $user = $this->userRepository->findOneBy(['id' => $id]);
```

Comment:
move this before the if and then your data will be loaded in the user already because of the `createForm` method. you don't need the `getData`